react-web/src/pages/subscription/payment/Payment.tsx

```diff
@@ -0,0 +1,176 @@
+        errorMsg = errorObj.info + ' Please try again.'
+    } else if (errorObj?.response?.message) {
+        errorMsg = errorObj?.response.message + ' Please try again.'
+    } else if (errorObj?.response?.data) {
+        errorMsg = errorObj.response.statusText + ' - ' + errorObj.response.data 
+    } else {
+      errorMsg = 'Something wrong occurred. Please try again later.'
+    }
+    setShowError(errorMsg);
+    const timeout = setTimeout(() => { 
+      clearTimeout(timeout); 
+      setShowError("") 
+    }, 5000)
+    setOpenModal(false)
+    console.log(errorMsg)
+}
+
+  useEffect(() => {
+    error && handleError(error)
+  }, [error])
+
+  const triggerPayment = async () => {
+    setShowError("");
+    setProcessing(true);
+    fetchData.get("/profile/current/balance").then((response) => {
+      const availableProfileBalance: number = response.data;
+      if (availableProfileBalance - currentTierPrice < 0) {
+        const min_fee = 1000000; // 1 ADA in lovelaces - min req for cardano wallet txn
+        const fee: BigNum = currentTierPrice > min_fee ? currentTierPrice : min_fee;
+        triggerTransactionFromWallet(fee);
+      } else {
+        fetchCurrentSubscription()
+      }
+    });
+  };
+
+  const triggerTransactionFromWallet = async (fee_in_lovelace: BigNum) => {
+    const response = await dispatch(payFromWallet({fee: fee_in_lovelace, wallet: wallet, address: address}))
+    if (response.payload) {
+      setTransactionId(response.payload);
+      fetchCurrentSubscription(true)
+    } else if (response?.error?.message) {
+      handleError(response.error.message);
+    }
+  }
+
+  const fetchCurrentSubscription = (isAfterPayment?: boolean) => {
+    fetchData.get("/profile/current/subscriptions").then((response: any) => {
+      const current = response.data.find((item: Subscription) => item.id === currentSubscriptionId)
+      if (current.tierId === state.id) {
+        if (current.status === 'pending') {
+          if (isAfterPayment) {
+            // keep calling this GET in a 1 sec delay to check the status
+            const timeout = setTimeout(() => {
+              clearTimeout(timeout)
+              fetchCurrentSubscription(isAfterPayment)
+            }, 1000)
+          } else {
+            currentTierPrice = current.price;
+            triggerPayment()
+          }
+        } else if (current.status === 'active' && !dayjs().isAfter(dayjs(current.endDate))) {
+          // payment retrieved from balance
+          setProcessing(false);
+          setOpenModal(true);
+        } 
+      }
+    }).catch(handleError);
+  }
+
+  const postSubscription = () => {
+    setShowError("");
+    setProcessing(true);
+    fetchData
+      .post("/profile/current/subscriptions/" + state.id)
+      .then((response: any) => {
+        currentSubscriptionId = response.data.id
+        fetchCurrentSubscription()
+      }).catch(handleError);
+  }
+
+  useEffect(() => {
+    if (!state) {
+      navigate(-1)
+    }
+  })
```

Comment:
I'm not sure I understand this. You run this effect at every render, and navigte back if `state` is false | null | 0 etc

1st question: when state goes null?
2nd question: why note adding normal dependencies `[state,navigate]` ?

Reply:
Removed this, as not needed